<!-- ITU- Control and Automation Engineering -->
<!-- KON-414E Introduction to Robot Autonomy -->
<!--      2024-2025 Term Final project       -->

<launch>
    <arg name="robot_namespace" default="/"/>

    <!-- Start Clearpath Inspeciton Map with Scout v2 Inside -->
    <include file="$(find intro_to_robot_autonomy)/launch/inspection_empty_world.launch"></include>


    <!-- Since robot model does not have direct wheel odom. We need to generate it -->
    <include file="$(find project1)/launch/launcher.launch"></include>
    <!-- <node pkg="intro_to_robot_autonomy" type="pub_odom_from_robot_state" name="pub_odom"/> -->

    <node pkg="intro_to_robot_autonomy" type="parametric_velocity_sequential" name="velocity_sequential_node"></node>

    <!-- Static transform frame  -->
    <node pkg="tf" type="static_transform_publisher" name="world_to_odom" args="-0.67825673 -0.631779108 0 0 0 0.638459 0.7696558 world odom 100"/>



    <node pkg="robot_localization" type="ekf_localization_node" name="ekf_localization_local">
        <rosparam command="load" file="$(find intro_to_robot_autonomy)/config/gps_confs/imu_wheel.yaml" />
        <remap from="odometry/filtered" to="odometry/filtered_local" />
    </node>


    <node pkg="robot_localization" type="ekf_localization_node" name="ekf_localization_global">
        <rosparam command="load" file="$(find intro_to_robot_autonomy)/config/gps_confs/local_gps.yaml" />
        <remap from="odometry/filtered" to="odometry/filtered_global" />
    </node>

    <node pkg="robot_localization" type="navsat_transform_node" name="navsat_transform_node" respawn="true">
        <param name="magnetic_declination_radians" value="0"/>
        <param name="yaw_offset" value="0"/>
        <param name="wait_for_datum" value="false"/>
        <param name="zero_altitude" value="true"/>
        <param name="publish_filtered_gps" value="true"/>
        <remap from="/imu/data" to="/scout/imu/data" />
        <remap from="/gps/fix" to="/scout/gps/fix" />
        <remap from="/odometry/filtered" to="/odometry/filtered_global" />
    </node>

    <node pkg="rqt_multiplot" type="rqt_multiplot" name="rqt_multiplot" output="screen" args="--multiplot-config /home/veli/scout_ws/src/intro_to_robot_autonomy/rqt_multiplot_conf/ground_truth_vs_local_vs_global.xml --multiplot-run-all">
    </node>



</launch>
